### ISSUE TITLE
BRAT-cIS (Capacity Inference System)

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/pyBRAT#35]
This is from #29 

> Joe, thought that it would be useful to have a BRAT calculator that would allow the user to run a simple inference system and get a capacity output “in the field.”
-NW: Yup, should be OK

## What it is:
I think we need to build out what we'll call the BRAT-cIS (capacity inference system) and distinguish that from the 'model' of BRAT-cFIS (capacity fuzzy inference system). This is nothing more than the combination of the two inference systems that comprise the capacity model of BRAT from the Macfarlane et al. ([2015](https://www.researchgate.net/publication/285590037_Modeling_the_capacity_of_riverscapes_to_support_beaver_dams)). What is different is we don't make the inference system fuzzy, and just run it from the two rule tables based on categorical inputs from the user. The first assesses capacity solely on vegetation next to stream, vs. within a 100 m of stream (on either side... but could break into river right and river left):
![table2](https://user-images.githubusercontent.com/9723905/39900218-8e3527d6-547d-11e8-933e-c25d82dcdca7.png)
This output is analagous to BRAT's `oVC_EX` [output](http://brat.riverscapes.xyz/Documentation/10-WorkingWithBRATOutputs.html).

The second takes that output, and feeds it in with a few other questions:
![table3](https://user-images.githubusercontent.com/9723905/39900227-9b3e7c70-547d-11e8-97eb-ffbc23896014.png)

Unlike the FIS, the output of  IS NOT a continuous number (in dams/km) but rather a category of Maximum Dam Density:

![legend_brat_damdensity_wide](https://user-images.githubusercontent.com/9723905/39900278-eccd8b94-547d-11e8-9c9a-67841530bba7.png)
This output is analagous to BRAT's `oCC_EX` [output](http://brat.riverscapes.xyz/Documentation/10-WorkingWithBRATOutputs.html) but is actually directly comparable to the `Ex_Category` output.

## The Specific Request

I think one of the Data Capture Apps (probably a ArcSurvey123) would be similar to what Nick has already made (https://www.youtube.com/watch?v=26rqpWeay-o) but is set up so it can be run either from Desktop or Field (i.e. #31), so that the call could be based on field data or desktop call. We just need to know who is making the call, where they are making it, and what evidence they are basing it off of.

### The User Inputs

The first part should be called the 'Vegetation to Support Beaver Dam Building'. Right now it looks like:
![brat_veg](https://user-images.githubusercontent.com/9723905/39900364-9dcfb890-547e-11e8-9860-b1d4df2dafb1.png)
That's pretty much there...

The second part brings in the 'Combined Capacity to Support Dam Building'. Here we need the following questions:
1. Can beaver build a dam at baseflows?
  - Probably can build dam
  - Can build dam
  - Can build dam (saw evidence of recent dams)
  - Could build dam at one time (saw evidence of relic dams)
  - Cannot build dam (streampower really high)
2. If beavers build a dam, consider what happens to the dam(s) in a typical flood (e.g. mean annual flood)?
  - Blowout
  - Occasional Blowout
  - Occasional Breach
  - Dam Persists
What empirical evidence exists of this at that location:
- None
- 
3. How does the reach slope impact their ability or need to build dams?
  - So steep they cannot build a dam (e.g. > 20% slope)
  - Probably can build dam
  - Can build dam (inferred)
  - Can build dam (evidence or current or past dams)
 -  Really flat (can build dam, but might not need as many as one dam might back up water > 0.5 km)

This stuff needs to expand the lower part of @webernick79's mock form
![brat_rest](https://user-images.githubusercontent.com/9723905/39900365-9fa89fba-547e-11e8-9997-c3f85de072b5.png)

Some additional contextual component should be:
- Streamflow
  - Perrenial
  - Potentially Intermitent
  - Intermitent
  - Potentially Ephemeral
  - Ephemeral
- If 'Intermittent' - also get 'Proximity to Perennial Water Source'
  - `>` 5 km
 - 1 - 5 km
 - < 1 km
- Channel Setting
  - Only Channel
  - Primary Anabranch
  - Secondary Anabranch or Side Channel 
  - Backwater

- Current Activity
  - Signs of Beaver Activity (checkboxes)
   - Current
   - Recent
   - Relic
  - Type of Beaver Activity
    - Dam Building
    - Dam Maineance
    - Food Caching
    - Woody Material Harvest
  - Existing Beaver Dams
   - How many primary 
   - How many secondary
   - How many complexes
   - How active (radio):
      - All maintained
      - Some maintaned
      - Some maintenance but all intact
      - Some maintenance - mixed intact, breached, blownout
      - No maintenance - but all intact
      - No maintenance - mixed intact, breached, blownout
      - No maintenance - all breached or blownout  

### The Output
- The output would be the polyline segment (250 m to 500 m long) of the stream where the assessment was made. The output would be the category of dam capacity (none, rare, occasional, frequent, or pervasive). It would have all of the above fields with having `idc` prefixes (for input data capture instead of `i` for inputs to model) and outputs having `odc` prefix. 

## Analyses it Could Support
We'll need to work with @banderson1618 and @bangen on these. A few that come to mind:
- Comparison of `odc_cIS` to `Ex_Category`,  on a reach by reach basis and having a map output with categories of:
  - Agreement
  - Minor Model Underprediction
  - Major Model Underprediction
  - Minor Model Overpredictino
  - Major Model Overprediction
- A pie chart (with percentages) and bar plot (with distances of stream) of above.
- An error matrix table
- An actual dam count 
   - Total
   - Primary
   - Secondary
   - Complexes

**Comment by joewheaton** [comment]
Do we want to ask whether or not this is an assessment of 'existing capacity' or 'potential capacity'?

**Comment by joewheaton** [comment]
Okay... Weber, check out #42 for forms (one basic, one advanced)

**Comment by joewheaton** [comment]
Note.... progress was tracked on #42 

@webernick79 

Please post link to finished BRAT_cIS Basic & BRAT_cIS Advanced when done here.

**Comment by joewheaton** [comment]
Read me

**Comment by webernick79** [comment]
@joewheaton and @wally-mac and @bangen 

OK, I think this is where I'm supposed to be posting related to BRAT field inference forms.

Here is a link to the [BRAT Field Assessment Form - Advanced](https://survey123.arcgis.com/share/f1c37b80ba3f482fbadb38251c5d7a7f?open=menu)

If I understand this correctly, this is also the form that Wally and Sara will be using for validation during their Truckee field visits.

At this point I don't think it's worth developing additional ESRI mobile functionality for the Truckee validation visits. I'm guessing you guys will also have iPads with GISkit that you will primarily be relying on. However, I am open to putting together a map in Arc Explorer or something like that as well.

I'll give this form a good going over tomorrow and continue to work out the kinks.

**Comment by joewheaton** [comment]
Thanks a ton @webernick79 for this!

I've tested the form on both a browser (desktop) and Survey123 (Android). Everything looks great. I think we should run with this for now and hold-off on tweaks until after we've used it in a real setting. Really nice job.  Very useful. 

I'm closing this as I think its done. Any new feature requests or tweaks can refer back to this, but should just be there own tickets.
